lib/rubocop/cop/internal_affairs/method_name_end_with.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module InternalAffairs
+      # This cop checks potentially usage of method identifier predicates
+      # defined in rubocop-ast instead of `method_name.end_with?`.
+      #
+      # @example
+      #   # bad
+      #   node.method_name.to_s.end_with?('=')
+      #
+      #   # good
+      #   node.assignment_method?
+      #
+      #   # bad
+      #   node.method_name.to_s.end_with?('?')
+      #
+      #   # good
+      #   node.predicate_method?
+      #
+      #   # bad
+      #   node.method_name.to_s.end_with?('!')
+      #
+      #   # good
+      #   node.bang_method?
+      #
+      class MethodNameEndWith < Base
+        include RangeHelp
+        extend AutoCorrector
+
+        MSG = 'Use `%<method_name>s` instead of `%<method_suffix>s`.'
+
+        # @!method method_name_end_with?(node)
+        def_node_matcher :method_name_end_with?, <<~PATTERN
+          {
+            ({ send csend }
+              ({ send csend }
+                $(... :method_name) :to_s) :end_with?
+              $(str { "=" "?" "!" }))
+            ({ send csend }
+              $(... :method_name) :end_with?
+            $(str { "=" "?" "!" }))
+          }
+        PATTERN
+
+        def on_send(node)
+          method_name_end_with?(node) do |method_name_node, end_with_arg|
+            range = range(method_name_node, node)
+            message = format(
+              MSG,
+              method_name: suggest_method_for_suffix[end_with_arg.value],
+              method_suffix: range.source
+            )
+
+            add_offense(node, message: message)
+          end
+        end
+        alias on_csend on_send
+
+        private
+
+        def range(method_name_node, node)
+          if method_name_node.call_type?
+            range_between(method_name_node.loc.selector.begin_pos, node.source_range.end_pos)
+          else
+            range_between(method_name_node.source_range.begin_pos, node.source_range.end_pos)
+          end
```

Comment:
Can you tweak it? It's more DRY.

```suggestion
          range = if method_name_node.call_type?
                    method_name_node.loc.selector
                  else
                    method_name_node.source_range
                  end

          range_between(range.begin_pos, node.source_range.end_pos)
```